metal/end_model/loss.py

```diff
@@ -20,14 +20,19 @@ class SoftCrossEntropyLoss(nn.Module):
 
     def __init__(self, weight=None, reduction="elementwise_mean"):
         super().__init__()
-        assert weight is None or isinstance(weight, torch.FloatTensor)
         self.weight = weight
+        if self.weight is not None:
+            self.weight = torch.Tensor(self.weight)
```

Comment:
What's the rationale for not explicitly forming a `FloatTensor` here?  Wouldn't that be better?